396 upgrade log fingerprints

michael-redpanda opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/redpanda-data/paramiko?pullRequest=1) <br/>All committers have signed the CLA.

michael-redpanda committed:
Switch debug logging to use .fingerprint

Per #396, FIPS mode rejects .get_fingerprint() because it
uses MD5.

The .fingerprint property uses SHA256, which is FIPS-compatible.

michael-redpanda committed:
Remove hexlify() calls since .fingerprint is str

michael-redpanda committed:
Add CHANGELOG entry
